pylearn2/termination_criteria/__init__.py

```diff
@@ -221,16 +221,16 @@ def initialize(self, model):
         if self._new_epochs:
             self._epochs_done = 0
```

Comment:
Actually, if `new_epochs` is True, then it does not matter how `push_monitor` was called, and it will work as expected.
If `new_epochs` is False, then the result would be different.